plugins/hls-explicit-fixity-plugin/src/Ide/Plugin/ExplicitFixity.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_GHC -Wall #-}
+{-# OPTIONS_GHC -Wno-deprecations #-}
+{-# OPTIONS_GHC -Wno-unticked-promoted-constructors #-}
+{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
+{-# HLINT ignore "Functor law" #-}
+
+module Ide.Plugin.ExplicitFixity where
+
+import           Control.Monad                 (forM)
+import           Control.Monad.IO.Class        (liftIO)
+import           Data.List.Extra               (nubOn)
+import qualified Data.Map                      as M
+import           Data.Maybe
+import qualified Data.Text                     as T
+import           Development.IDE               hiding (pluginHandlers)
+import           Development.IDE.GHC.Compat
+import           Development.IDE.Spans.AtPoint (pointCommand)
+import           Ide.PluginUtils               (getNormalizedFilePath,
+                                                handleMaybeM, pluginResponse)
+import           Ide.Types                     hiding (pluginId)
+import           Language.LSP.Types
+
+pluginId :: PluginId
+pluginId = "explicitFixity"
+
+descriptor :: PluginDescriptor IdeState
+descriptor = (defaultPluginDescriptor pluginId)
+    { pluginHandlers = mkPluginHandler STextDocumentHover hover
+    }
+
+hover :: PluginMethodHandler IdeState TextDocumentHover
+hover state plId (HoverParams (TextDocumentIdentifier uri) pos _) = pluginResponse $ do
+    nfp <- getNormalizedFilePath plId uri
+    har <- handleMaybeM "Unable to get HieResult"
+        $ liftIO
+        $ runAction "ExplicitFixity.HieResult" state
+        $ use GetHieAst nfp
```

Comment:
This sequence sure does come up a lot :laughing:

Reply:
Since our new plugins turn to `ExceptT`, these boilerplate codes are ubiquitous...

Reply:
I promise when I get some time I plan on tackling this 😃